### PR TITLE
Add High-Availability and Read Replicas Support for PostgreSQL

### PR DESCRIPTION
**Pull request by Xtigyro**
- Externalize the creation of PostgreSQL backend using the official PostgreSQL Helm chart.
- Add high-availability and performance read replicas support for PostgreSQL.
- Add storage persistence for PostgreSQL.
- Allow only manual storage deletion for PostgreSQL.
- Tweak r10k cronjob schedules and Puppet Server components storage size.
- Add Hiera v5 example with explanations.
- Improve informational comments in values file.
- Bump Puppet Server to `v6.12.1`, PuppetDB to `v6.12.0`, and PostgreSQL to `v11.7.*`.
- Add `CODEOWNERS`.
- Numerous other small tweaks.

@underscorgan @Iristyle Please use the `Squash and merge` button upon merging.

CC: @slconley @scottcressi @mwaggett @nwolfe @adrienthebo @dhollinger @raphink @binford2k

### CODE REVIEW

**Comment by Xtigyro** [comment]
@underscorgan @Iristyle This should be ready for merging.

CC: @slconley @scottcressi @mwaggett @nwolfe @adrienthebo @dhollinger @raphink @binford2k

File: requirements.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+dependencies:
+
+  - name: postgresql
+    version: "9.4.*"
```
Comment (underscorgan) [inline requirements.yaml:4]: wondering how much testing you've done with this? I know we run against 9.6 currently, wasn't sure how things would work with 9.4

**Review by underscorgan — COMMENTED** [review]
one question about postgres version, but otherwise looks good to me!

File: requirements.yaml (hunk 1 above)
Comment (Xtigyro) [reply requirements.yaml:4]: This is the version of the chart - it's actually `v11.9.*`. And yeah - it's OK.

Comment (Xtigyro) [reply requirements.yaml:4]: I've tested most use cases - incl. also running multiple PostgreSQL Read Replicas.

**Review by underscorgan — APPROVED** [review]

**Pull request merged**